packages/ui/src/components/alert/AlertRoot.tsx

```diff
@@ -0,0 +1,140 @@
+import { forwardRef, PropsWithChildren, useLayoutEffect, useRef, useState } from 'react'
+
+import { Button } from '@components/button'
+import { Icon, IconNameMap } from '@components/icon'
+import { useResizeObserver } from '@hooks/use-resize-observer'
+import { cn } from '@utils/cn'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+const alertVariants = cva('cn-alert', {
+  variants: {
+    theme: {
+      info: 'cn-alert-info',
+      danger: 'cn-alert-danger',
+      warning: 'cn-alert-warning'
+    }
+  },
+  defaultVariants: {
+    theme: 'info'
+  }
+})
+
+const iconMap: Record<NonNullable<VariantProps<typeof alertVariants>['theme']>, keyof typeof IconNameMap> = {
+  info: 'info-circle',
+  danger: 'cross-circle',
+  warning: 'warning-triangle-outline'
+}
+
+const MAX_HEIGHT = 60
+
+export interface AlertRootProps extends PropsWithChildren<VariantProps<typeof alertVariants>> {
+  className?: string
+  dismissible?: boolean
+  onDismiss?: () => void
+  expandable?: boolean
+}
+
+export const AlertRoot = forwardRef<HTMLDivElement, AlertRootProps>(
+  ({ className, theme, children, dismissible, onDismiss, expandable }, ref) => {
+    const [isVisible, setIsVisible] = useState(true)
+    const [isExpanded, setIsExpanded] = useState(false)
+    const [isOverflowing, setIsOverflowing] = useState(false)
+    const contentRef = useRef<HTMLDivElement>(null)
+
+    const handleDismiss = () => {
+      setIsVisible(false)
+      onDismiss?.()
+    }
+
+    const iconName: keyof typeof IconNameMap = iconMap[theme ?? 'info']
+
+    const toggleExpand = () => setIsExpanded(prev => !prev)
+
+    const shouldShowButton = expandable && isOverflowing
+
+    useResizeObserver(
+      contentRef,
+      el => {
+        if (!el) return
+        const { scrollHeight } = el
+        setIsOverflowing(scrollHeight > MAX_HEIGHT)
+      },
+      100
+    )
+
+    useLayoutEffect(() => {
+      if (!expandable || !contentRef.current) return
+
+      const el = contentRef.current
+      setIsOverflowing(el.scrollHeight > MAX_HEIGHT)
+    }, [children, expandable])
+
+    if (!isVisible) return null
+
+    return (
+      <div ref={ref} role="alert" className={cn(alertVariants({ theme }), className)}>
+        {dismissible && (
+          <Button
```

Comment:
Can we add an `aria-label` to this since there's nothing for a screen reader to reference.

Reply:
Fixed